heat/core/operations.py

```diff
@@ -544,7 +549,620 @@ def sum(x, axis=None, out=None):
 
     return __reduce_op(x, torch.sum, MPI.SUM, axis, out)
 
-  
+
+def merge_means(mu1, n1, mu2, n2):
+    """
+    Function to merge two means by pairwise update
+
+    Parameters
+    ----------
+    mu1 : 1D ht.tensor or 1D torch.tensor
+          Calculated mean
+    n1 : 1D ht.tensor or 1D torch.tensor
+         number of elements used to calculate mu1
+    mu2 : 1D ht.tensor or 1D torch.tensor
+          Calculated mean
+    n2 : 1D ht.tensor or 1D torch.tensor
+         number of elements used to calculate mu2
+
+    Returns
+    -------
+    mean of combined set
+    number of elements in the combined set
+    """
+    if mu1.item() == np.nan:
+        mu1 = 0.0
+    if mu2.item() == np.nan:
+        mu2 = 0.0
+    delta = mu2.item() - mu1.item()
+    n1 = n1.item()
+    n2 = n2.item()
+    return mu1 + n2 * (delta / (n1+n2)), n1 + n2
+
+
+def mean(x, dimen=None, all_procs=False):
+    """
+    Calculates and returns the mean of a tensor.
+    If a dimension is given, the mean will be taken in that direction.
+
+    Parameters
+    ----------
+    x : ht.tensor
+        Values for which the mean is calculated for
+    dimen : None, Int, iterable
+            Dimension which the mean is taken in.
+            Default: None -> mean of all data calculated
+    all_procs : Bool
```

Comment:
all_procs seems to be an unnecessary flag. Heat has rules for the calculation of reduction operations across split axis. There are two major cases:

a) the reduction is performed across the split axis, then the result tensor is not split and a copy on each node
b) the reduction is performed along the split axis, the result tensor has the same split axis, all the nodes have their own partial chunk

Should be written down somewhere, agreed!

Reply:
okay my idea was that if at some time it was desired to have all of the means on one node then it could be done easily but if that is the case then i will simply remove the flag and the accompanying code